tests/backup/backup_helper.go

```diff
@@ -1242,7 +1246,377 @@ func backupSuccessCheck(backupName string, orgID string, retryDuration time.Dura
+						errors = append(errors, err)
+					}
+
+					if backedupVol.StorageClass != *pvcObj.Spec.StorageClassName {
+						err := fmt.Errorf("the Storage Class of the volume as per the backup [%s] is [%s], but the one found in the scheduled namesapce is [%s]", backedupVol.GetName(), backedupVol.StorageClass, *pvcObj.Spec.StorageClassName)
+						errors = append(errors, err)
+					}
+
+					continue volloop
+				}
+			}
+
+			// The following error means that something WAS not backed up, OR it wasn't supposed to be backed up, and we forgot to exclude the check.
+			err = fmt.Errorf("the volume [%s] corresponding to PVC(name: [%s], namespace: [%s]) was present in the cluster corresponding to the PVC's context, but not in the backup [%s]", pvcObj.Spec.VolumeName, pvcObj.GetName(), pvcObj.GetNamespace(), backupName)
+			errors = append(errors, err)
+		}
+
+		for vol, found := range namespacedBackedUpVolumesFound {
+			if !found {
+				err := fmt.Errorf("volume (name: [%s], namespace: [%s]) in backup [%s], doesn't have a corresponding PVC spec in the context [%v]", vol.GetName(), vol.GetNamespace(), backupName, clusterAppsContextNamespace)
+				errors = append(errors, err)
+			} else {
+				log.Infof("volume (name: [%s], namespace: [%s]) in backup [%s] has a PVC spec in the context [%s]", vol.GetName(), vol.GetNamespace(), backupName, clusterAppsContextNamespace)
+			}
+		}
+	}
+
+	for vol, found := range backedUpVolumesFound {
+		if !found {
+			err := fmt.Errorf("volume (name: [%s], namespace: [%s]) in backup [%s], doesn't have a corresponding PVC spec in ANY provided context", vol.GetName(), vol.GetNamespace(), backupName)
+			errors = append(errors, err)
+		}
+	}
+
+	return
+}
+
+// GetBackupCtxsFromScheduledCtxs clones and returns the scheduled contexts
+// * after filtering its `spec`s to only include the resources that are in the backup.
+// *
+// * Returns:
+// * backupclusterAppsContexts: the filtered context (backup)
+// * ScheduledCtxNotFoundInBackupErrors: these are errors that are generated when namespaces and/or specObjs in the scheduled context are not found in the backup
+// * BackupNotFoundInAnyScheduledCtxErrors: these are errors that are generated when namespaces and/or resources in the backup context are not found in the scheduled contextx
+// * otherErrors: all the other kinds of errors; Don't ignore these are they're serious errors
+// *
+// * NOTES:
+// * - make sure to include Contexts of *all* namespaces which are supposed to contain the backup objects
+// * - The errors returned are all tolerable errors that have been caught until an intolerable error was encountered
+func GetBackupCtxsFromScheduledCtxs(backupInspectResponse *api.BackupInspectResponse, scheduledAppContexts []*scheduler.Context) (backupclusterAppsContexts []*scheduler.Context, ScheduledCtxNotFoundInBackupErrors, BackupNotFoundInAnyScheduledCtxErrors, otherErrors []error) {
```

Comment:
too many return args just for errors. just return error if there is an error. keep it simple